src/event.rs

```diff
@@ -1,4 +1,23 @@
-//! This module contains the `WindowEvent` type and re-exports glutin event
+//! This module contains the `Event` enum and re-exports glutin event
 //! types.
 
-pub use winit::{ElementState, Event, KeyboardInput, MouseButton, VirtualKeyCode, WindowEvent};
+pub use winit::{DeviceEvent, DeviceId, ElementState, KeyboardInput, MouseButton, VirtualKeyCode, WindowEvent};
+
+/// Events that the engine might send to the user.
+#[derive(Clone)]
+pub enum Event {
+    /// A WindowEvent from the winit crate
+    WindowEvent(WindowEvent),
+    /// A DeviceEvent from the winit crate
+    DeviceEvent(DeviceId, DeviceEvent),
+    /// Awakened event from the winit crate
+    Awakened,
+    // Add more events here as needed.
+    #[doc(hidden)]
```

Comment:
Could we please have docs before attributes? I think that's most used order in this crate and also the one I prefer.